### PR TITLE
"Add missing fields" quickfix: add trailing comma if needed

### LINKED ISSUE #8342: Trailing comma should be placed in the last field when creating type
When I type a struct's name, the plugin will suggest the struct's type and will fill the fields for me, without the values. This is great, expect for 1 tiny little thing: the missing trailing comma in the last field. e.g.:

```rust
let y = MyType { 
    a: 1,
    b: 2 <- missing comma
}
```

### PR DESCRIPTION
**Pull request by t-kameyama**
Fixes #8342

### CODE REVIEW

File: src/main/kotlin/org/rust/lang/core/psi/RsDefaultValueBuilder.kt [hunk 1]
```diff
@@ -153,6 +158,8 @@ class RsDefaultValueBuilder(
             structLiteral.addAfter(psiFactory.createNewline(), structLiteral.lbrace)
         }
 
+        structLiteral.addTrailingCommaIfNeeded()
```
Comment (dima74) [inline src/main/kotlin/org/rust/lang/core/psi/RsDefaultValueBuilder.kt:161]: I propose to make it a bit simplier - if struct is going to be multiline
```kotlin
val isMultiline = forceMultiLine || structLiteral.textContains('\n')
```
then add comma when adding last field (body of `if (addBefore == null)`)
```kotlin
if (isMultiline && fieldDecl == fieldsToAdd.last()) {
    structLiteral.addAfter(psiFactory.createComma(), added)
}
```

Comment (Undin) [reply src/main/kotlin/org/rust/lang/core/psi/RsDefaultValueBuilder.kt:161]: Please, do not add any heuristics for trailing comma
It's very confusing
We just need to use some code style settings. As I can see, there isn't a single style across different languages so let's use someone

Comment (dima74) [reply src/main/kotlin/org/rust/lang/core/psi/RsDefaultValueBuilder.kt:161]: I do not quite understand your proposal. Both our built-in formatter and rustfmt will add trailing comma in struct literals, so what is wrong if add it during code generation?

Comment (Undin) [reply src/main/kotlin/org/rust/lang/core/psi/RsDefaultValueBuilder.kt:161]: Oh, I see that we trailing comma indeed used by default in other places. Ok then, let's add it as proposed in this PR
But ideally, we need an option to turn this behaviour off

File: src/test/kotlin/org/rust/ide/annotator/fixes/AddStructFieldsFixTest.kt [hunk 2]
```diff
@@ -103,7 +103,7 @@ class AddStructFieldsFixTest : RsAnnotatorTestBase(RsExpressionAnnotator::class)
                 0: 92,
                 1: 0/*caret*/,
                 2: 92,
-                3: 0
+                3: 0,
```
Comment (dima74) [inline src/test/kotlin/org/rust/ide/annotator/fixes/AddStructFieldsFixTest.kt:106]: I think that we should add trailing comma only if we just generated the last field, and shouldn't touch the last field if it already exists and we generated some fields in the middle

Comment (t-kameyama) [reply src/test/kotlin/org/rust/ide/annotator/fixes/AddStructFieldsFixTest.kt:106]: [04e4535](https://github.com/intellij-rust/intellij-rust/pull/9515/commits/04e4535008cbe3c94f795c230597395597f7829e)

**Review by dima74 — APPROVED** [review]
Thanks!
Could you please squash commits

**Comment by t-kameyama** [comment]
> Could you please squash commits

Done

**Pull request merged**